src/commands/e2e/ConfigTemplate.ts

```diff
@@ -53,53 +54,77 @@ export class ConfigTemplate {
         let junitConfig = '';
         if (jUnitReportPath) {
             junitConfig = `, ['junit', {
-                outputDir: '${path.resolve(jUnitReportPath)}',
+                outputDir: '${WdioConfigGenerator.safePath(path.resolve(jUnitReportPath))}',
                 outputFileFormat:
                     function(opts) {
                         return \`e2e.xunit.\${opts.capabilities.browserName}.\${new Date().toISOString()}.xml\`;
                     }
                 }]`;
         }
 
-        this.template = `exports.config = {
-            before: function () {
-                var config = require('${e2eFolder}/tsconfig.json');
-                require('${mainRepoDir}/node_modules/tsconfig-paths').register({
-                   baseUrl: '${e2eFolder}',
-                   paths: config.compilerOptions.paths || []
-               });
-                require('${mainRepoDir}/node_modules/ts-node').register({
-                    files: true,
-                    project: '${e2eFolder}/tsconfig.json'
-                });
-            },
-            runner: 'local',
-            specs: [
-                '${e2eFolder}/specs/${specs || '**/*.spec.ts'}'
-            ],
-            exclude: [],
-            maxInstances: 1,
-            capabilities: ${capabilities},
-            logLevel: 'info',
-            bail: 0,
-            baseUrl: '${baseUrl}',
-            waitforTimeout: 100000,
-            connectionRetryTimeout: 90000,
-            connectionRetryCount: 3,
-            services: ['selenium-standalone', 'intercept'],
-            skipSeleniumInstall: ${noInstall ? 'true' : 'false'},
-            framework: 'jasmine',
-            jasmineNodeOpts: {
-                defaultTimeoutInterval: ${timeout}
-            },
-            plugins: {
-                webdriverajax: {}
-            },
-            ${ieDriver}
-            reporters: ['spec'
-            ${junitConfig}
-            ]
-        };`;
+        let screenshotConfig = '';
+        if (screenShotPath) {
+            screenshotConfig = `if (!test.passed) {
+            const screenshotDir = '${WdioConfigGenerator.safePath(path.resolve(screenShotPath))}';
+
+            if (!fs.existsSync(screenshotDir)) {
+                fs.mkdirSync(screenshotDir);
+            }
+
+            const filePath = path.join(
+                screenshotDir,
+                test.fullName.replace(/[^a-z0-9]/gi, '_').toLowerCase()
```

Comment:
I would add the date (maybe even time) to the start of the name (e.g. yyyy.mm.dd_testname.png) unless it automatically posts the sreenshots in a subdirectory with the date (similar to the other test results)